File: src/main/java/cc/moecraft/logger/format/AnsiFormat.java
package cc.moecraft.logger.format;

import static cc.moecraft.logger.format.AnsiConstants.*;
import static cc.moecraft.logger.format.AnsiConstants.PREFIX;
import static cc.moecraft.logger.format.AnsiConstants.SUFFIX;

/**
 * 此类由 Hykilpikonna 在 2018/05/04 创建!
 * Created by Hykilpikonna on 2018/05/04!
 * Github: https://github.com/hykilpikonna
 * QQ: admin@moecraft.cc -OR- 871674895
 *
 * @author Hykilpikonna
 */
public enum AnsiFormat 
{
    RESET("0"),

    HIGH_INTENSITY("1", "l"),
    LOW_INTENSITY("2"),
    ITALIC("3", "o"),
    UNDERLINE("4", "n"),
    BLINK("5"),
    RAPID_BLINK("6"),
    REVERSE_VIDEO("7"),
    INVISIBLE_TEXT("8");

    String code;

    AnsiFormat(String code, String ... placeholders)
    {
        this.code = code;
        formats.add(this);
        for (String placeholder : placeholders) formatsPlaceholderIndex.put(placeholder, this);
    }

    @Override
    public String toString() 
    {
        return PREFIX + code + SUFFIX;
    }
}

File: src/main/kotlin/org/hydev/logger/format/AnsiFormat.kt
package org.hydev.logger.format

import org.hydev.logger.format.AnsiConstants.ESC_PREFIX
import org.hydev.logger.format.AnsiConstants.SUFFIX

/**
 * 此类由 Hykilpikonna 在 2018/05/04 创建!
 * Created by Hykilpikonna on 2018/05/04!
 * Github: https://github.com/hykilpikonna
 * Meow!
 *
 * @author Hykilpikonna
 */
enum class AnsiFormat(var code: Int, vararg val placeholders: Char)
{
    RESET(0),
    HIGH_INTENSITY(1, 'l'),
    LOW_INTENSITY(2),
    ITALIC(3, 'o'),
    UNDERLINE(4, 'n'),
    BLINK(5),
    RAPID_BLINK(6),
    REVERSE_VIDEO(7),
    INVISIBLE_TEXT(8);

    override fun toString() = value
    val value = ESC_PREFIX + code + SUFFIX
}
